cpp/include/cudf/detail/null_mask.cuh

```diff
@@ -69,6 +69,10 @@ __global__ void offset_bitmask_binop(Binop op,
   constexpr auto const word_size{detail::size_in_bits<bitmask_type>()};
   auto const tid = threadIdx.x + blockIdx.x * blockDim.x;
 
+  size_type const last_bit_index  = source_size_bits - 1;
+  size_type const num_slack_bits  = word_size - (last_bit_index % word_size) - 1;
```

Comment:
This can be computed with `intra_word_index`, I think. Let's reuse functions and avoid doing the modular math by hand.